pkg/agent/agentserver/server.go

```diff
@@ -70,9 +70,14 @@ type ProxyServer struct {
 	backends map[string][]agent.AgentService_ConnectServer
 	agentIDs []string
 	random   *rand.Rand
+	// This interface makes testing easier.
+	BackendGetter BackendGetter
```

Comment:
I actually really like the way you do this. Perhaps this can be used not only for testing, but also when we want look at new strategies for load balancing. I'm wondering if it makes sense to add this interface as an optional parameter to the `agentserver.NewProxyServer` function.

Reply:
We can add the parameter when it's necessary.